src/crimson/osd/ops_executer.h

```diff
@@ -180,6 +180,35 @@ class OpsExecuter : public seastar::enable_lw_shared_from_this<OpsExecuter> {
   size_t num_write = 0;   ///< count update ops
 
   SnapContext snapc; // writer snap context
+  struct CloningContext {
+    SnapSet new_snapset;
+    pg_log_entry_t log_entry;
+
+    void apply_to(
+      const eversion_t& at_version,
+      std::vector<pg_log_entry_t>& log_entries,
+      ObjectContext& processed_obc) &&;
+  };
+  std::unique_ptr<CloningContext> cloning_ctx;
+
+  std::unique_ptr<CloningContext> execute_clone(
+    const SnapContext& snapc,
+    const ObjectState& initial_obs,
+    const SnapSet& initial_snapset,
+    PGBackend& backend,
+    ceph::os::Transaction& txn);
+
+  static bool should_clone(const ObjectContext& initial_obc,
+                           const SnapContext& snapc) {
```

Comment:
Let's be consistent about param list indentation.  Change to match execute_clone above (also elsewhere in this PR).